src/navigator-card-body.jsx

```diff
@@ -0,0 +1,302 @@
+    const isMounted = useRef(null);
+
+    function calculateBoxPerRow () {
+        const boxes = document.querySelectorAll(".item-button");
+        if (boxes.length > 1) {
+            let i = 0;
+            const total = boxes.length;
+            const firstOffset = boxes[0].offsetTop;
+            while (++i < total && boxes[i].offsetTop === firstOffset);
+            setBoxPerRow(i);
+        }
+    }
+
+    const onDoubleClickNavigate = useCallback((file) => {
+        const newPath = [...path, file.name].join("/");
+        if (file.type === "dir" || file.to === "dir") {
+            setHistory(h => [...h.slice(0, historyIndex + 1), [...path, file.name]]);
+            setHistoryIndex(h => h + 1);
+
+            cockpit.location.go("/", { path: encodeURIComponent(newPath) });
+        }
+    }, [
+        path,
+        historyIndex,
+        setHistoryIndex,
+        setHistory
+    ]);
+
+    useEffect(() => {
+        calculateBoxPerRow();
+        window.onresize = calculateBoxPerRow;
+        return () => {
+            window.onresize = undefined;
+        };
+    });
+
+    useEffect(() => {
+        const onKeyboardNav = (e) => {
+            if (e.key === "ArrowRight") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = selectedIdx < sortedFiles.length - 1
+                        ? selectedIdx + 1
+                        : 0;
+
+                    return [sortedFiles[newIdx]];
+                });
+            } else if (e.key === "ArrowLeft") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = selectedIdx > 0
+                        ? selectedIdx - 1
+                        : sortedFiles.length - 1;
+
+                    return [sortedFiles[newIdx]];
+                });
+            } else if (e.key === "ArrowUp") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = Math.max(selectedIdx - boxPerRow, 0);
+
+                    return [sortedFiles[newIdx]];
+                });
+            } else if (e.key === "ArrowDown") {
+                setSelected(_selected => {
+                    const firstSelectedName = _selected?.[0]?.name;
+                    const selectedIdx = sortedFiles?.findIndex(file => file.name === firstSelectedName);
+                    const newIdx = Math.min(selectedIdx + boxPerRow, sortedFiles.length - 1);
+
+                    return [sortedFiles[newIdx]];
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)